test/js_test.go

```diff
@@ -5224,6 +5250,50 @@ func TestJetStreamSubscribe_RateLimit(t *testing.T) {
 	}
 }
 
+func TestJetStreamSubscribe_FilterSubjects(t *testing.T) {
+	s := RunBasicJetStreamServer()
+	defer shutdownJSServerAndRemoveStorage(t, s)
+
+	nc, js := jsClient(t, s)
+	defer nc.Close()
+
+	var err error
+
+	// Create the stream using our client API.
+	_, err = js.AddStream(&nats.StreamConfig{
+		Name:     "TEST",
+		Subjects: []string{"foo", "bar", "baz"},
+	})
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	for i := 0; i < 5; i++ {
+		js.Publish("foo", []byte("msg"))
+	}
+	for i := 0; i < 5; i++ {
+		js.Publish("bar", []byte("msg"))
+	}
+	for i := 0; i < 5; i++ {
+		js.Publish("baz", []byte("msg"))
+	}
+
+	sub, err := js.SubscribeSync("", nats.BindStream("TEST"), nats.ConsumerFilterSubjects("foo", "baz"))
```

Comment:
is this an ephemeral consumer? do we need another one with a durable name to cover some of the conditions above?